interp/src/flatten/structures/environment/env.rs

```diff
@@ -248,6 +249,63 @@ impl<C: AsRef<Context> + Clone> Environment<C> {
         self.ctx.as_ref()
     }
 
+    pub fn iter_cells(
+        &self,
+    ) -> impl Iterator<Item = (String, Vec<String>)> + '_ {
+        let env = self;
+        let cell_names = self.cells.iter().map(|(idx, _ledger)| {
+            (idx.get_full_name(env), self.iter_cells_port_helper(idx))
+        });
+
+        cell_names
+        // get parent from cell, if not exist then lookup component ledger get base idxs, go to context and get signature to get ports
+        // for cells, same thing but in the cell ledger, subtract child offset from parent offset to get local offset, lookup in cell offset in component info
+    }
+
+    //not sure if beneficial to change this to be impl iterator as well
+    fn iter_cells_port_helper(&self, cell: GlobalCellIdx) -> Vec<String> {
```

Comment:
I would make this name a little clearer but otherwise seems fine.  Could also be made an internal function of `iter_cells` which would be better.